### PR TITLE
Add a public build script

### PR DESCRIPTION
**Pull request by brycekbargar**
### TLDR;

Previously our build process was entirely Team Foundation Server based, using the TFS build task for building, testing, and creating a rap file.

Since this is going to continue life as an open source project we need a way for anyone to be able to build the project and create a rap file.

The plan is to setup a new TFS build definition that uses this same script for what was previously being done with build tasks.

### There be monsters

We're relying on an "alpha" nuget package from Relativity for building the RAP file that isn't super well documented. This is so that we can keep our corporate rap builder closed source.

We've temporarily lost code coverage numbers for this project. They are going to be added later when the TFS build definition is created.

### CODE REVIEW

File: Deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+.\build.ps1 CreateRap -Version 9.6.0.0
+.\build.ps1 -TaskList CreateRap 
+.\build.ps1 -TaskList CreateRap 9.6.0.0
+.\build.ps1 -TaskList CreateRap -Version 9.6.0.0
+```
+
+You can find the output rap in the Deployment/Packages folder.
+*Note:* the output rap will override any other raps with the same version.
+
+## Windows Subsystem for Linux
```
Comment (Arithmomaniac) [inline Deployment/README.md]: I don't think we should include WSL stuff. It adds confusion for the other 99% of people and can lead people to think this builds cross-platform.

File: Deployment/default.ps1 [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	$scriptrunner_rap = "$Deployment\ScriptRunner.rap"
+
+	$Version = $null
+}
+
+Task Default -Depends UnitTest
+
+Task PackageBuild -Depends RestorePackages {
+	Exec { 
+		msbuild "$sln" `
```
Comment (Arithmomaniac) [inline Deployment/default.ps1]: Surprised you don't need to specify a publish profile for the custom page

Comment (brycekbargar) [reply Deployment/default.ps1]: Yeah. I was surprised too but the artifacts are contained in the output in a _PublishedWebsites folder. I think there is some setting on one of the csproj files that takes care of everything.
